source/backend/discovery/test/getAllSdkResources.js

```diff
@@ -1547,6 +1554,77 @@ describe('getAllSdkResources', () => {
                     ]
                 });
             });
+            describe(AWS_DYNAMODB_STREAM, () => {
+
+                it('should discover DynamoDB Streams', async () => {
+                    const schema = require('./fixtures/relationships/dynamodb/stream.json');
+                    const {table, stream} = generate(schema);
+
+                    const mockDynamoDBStreamsClient = {
+                        createDynamoDBStreamsClient(credentials, region) {
+                            return {
+                                async describeStream(streamArn) {
+                                    if(credentials.accessKeyId === ACCESS_KEY_X && region === EU_WEST_2) {
+                                        return { StreamARN: stream.arn }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    const arn = `arn:aws:dynamodb:${EU_WEST_2}:${ACCOUNT_X}:table/test/stream`;
+
+                    const actual = await getAllSdkResources({...mockAwsClient, ...mockDynamoDBStreamsClient}, [table]);
+
+                    const actualDynamoDBStreamResource = actual.find(x => x.arn === arn);
+
+                    assert.deepEqual(actualDynamoDBStreamResource, {
+                        id: arn,
+                        accountId: ACCOUNT_X,
+                        awsRegion: EU_WEST_2,
+                        availabilityZone: NOT_APPLICABLE,
+                        arn: arn,
+                        resourceId: arn,
+                        resourceName: arn,
+                        resourceType: AWS_DYNAMODB_STREAM,
+                        relationships: [],
+                        configuration: {
+                            StreamARN: "arn:aws:dynamodb:eu-west-2:xxxxxxxxxxxx:table/test/stream"
+                        },
+                        configurationItemStatus: "ResourceDiscovered",
+                        tags: []
+                    });
+
+                });
+
+            });
+
+            describe(AWS_DYNAMODB_TABLE, () => {
+
+                it('should discover DynamoDB Tables without streams', async () => {
+                    const schema = require('./fixtures/relationships/dynamodb/table.json');
+                    const {tableNoStream} = generate(schema);
+
+                    const arn = `arn:aws:dynamodb:${EU_WEST_2}:${ACCOUNT_X}:table/test`;
+
+                    const actual = await getAllSdkResources({...mockAwsClient}, [tableNoStream]);
```

Comment:
Let's test that the `actual` array only has a size of 1 to ensure that no stream resource was created:

```js
assert.lengthOf(actual, 1);
```